src/widgets/auto-style/style-utils.js

```diff
@@ -1,67 +1,76 @@
 var _ = require('underscore');
+var postcss = require('postcss');
 
-var ATTR_TO_NUMBER = {
-  'marker-fill': 1,
-  'line-color': 2,
-  'polygon-fill': 3
-};
-
-function getAttrRegex (attr, multi) {
-  return new RegExp('\\' + 's' + attr + ':.*?(;|\n)', multi ? 'g' : '');
-}
+function isPropertyIncluded (cartocss, attr) {
+  var cssTree = postcss().process(cartocss);
+  var root = cssTree.result.root;
+  var propertyIncluded = false;
 
-function removeEmptyLayer (cartocss) {
-  return cartocss.replace(/[^;}{]*{((\s|\n)*?)}/g, '');
-}
-
-function setFlagInCartocss (cartocss, attr, flag) {
-  var exist = cartocss.search(getAttrRegex(attr, false)) >= 0;
+  root.walk(function (node) {
+    var parentNode = node.parent;
+    if (node.type === 'decl' && node.prop === attr) {
+      if (!isSelectorRule(parentNode) || isMapnikGeometrySelectorRule(parentNode)) {
```

Comment:
Ahh, now I remember why I included this. We should consider that the property is defined if it is inside a mapnik geometry selector. If it is not inside a selector-rule (considitional) or if it is inside a selector rule but it is defined by the mapnik-geometry thing.